app/models/concerns/track_activity.rb

```diff
@@ -55,6 +57,31 @@ def project_activity(activity)
     namespace_project_sample_activity_parameters(params, activity)
   end
 
+  def group_activity(activity) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    activity_trackable = activity_trackable(activity, Group)
+
+    base_params = {
+      created_at: activity.created_at.strftime(
+        I18n.t('time.formats.abbreviated')
+      ),
+      key: "activity.#{activity.key}_html",
+      user: activity_creator(activity),
+      group: activity_trackable,
+      created_group: get_object_by_id(activity.parameters[:created_group_id], Group),
+      transferred_group: get_object_by_id(activity.parameters[:transferred_group_id], Group),
+      removed_group_puid: activity.parameters[:removed_group_puid],
+      project: get_object_by_id(activity.parameters[:project_id], Project),
+      project_puid: activity.parameters[:project_puid],
+      name: activity_trackable.name,
+      type: 'Namespace',
+      action: activity.parameters.key?(:action) ? activity.parameters[:action] : 'default'
+    }
+
+    return base_params if activity.parameters[:action].blank?
+
+    transfer_activity_parameters(base_params, activity)
+  end
```

Comment:
**suggestion (non-blocking):** There is a lot going on here, and we always keep adding the `rubocop:disabled` comment.  Can you break this up a bit like:

```ruby
def group_activity(activity)
  activity_trackable = activity_trackable(activity, Group)

  base_params = build_base_params(activity, activity_trackable)
  additional_params = build_additional_params(activity)

  params = base_params.merge(additional_params)
  
  return params if activity.parameters[:action].blank?

  transfer_activity_parameters(params, activity)
end

private

def build_base_params(activity, activity_trackable)
  {
    created_at: format_created_at(activity.created_at),
    key: "activity.#{activity.key}_html",
    user: activity_creator(activity),
    group: activity_trackable,
    name: activity_trackable.name,
    type: 'Namespace',
    action: activity.parameters.fetch(:action, 'default')
  }
end

def build_additional_params(activity)
  {
    created_group: get_object_by_id(activity.parameters[:created_group_id], Group),
    transferred_group: get_object_by_id(activity.parameters[:transferred_group_id], Group),
    removed_group_puid: activity.parameters[:removed_group_puid],
    project: get_object_by_id(activity.parameters[:project_id], Project),
    project_puid: activity.parameters[:project_puid]
  }
end

def format_created_at(created_at)
  created_at.strftime(I18n.t('time.formats.abbreviated'))
end
```

Reply:
Refactored in 40309cfa83dfb1820548f4e73e858e3d2a05d3dc to remove most of the rubocop disables